### ISSUE TITLE
Periodic hangs during training/analysis

### ISSUE THREAD

**Issue by rglanz** [DeepLabCut/DeepLabCut#299]
**Your Operating system and DeepLabCut version**

Windows 10, Anaconda environment , DLC 2.0.6.3

**Describe the problem**

Process will occasionally hang with no progress being made during analysis or training. E.g. I will start training a network or analyzing a video at night, and check the progress in the morning and it will have been stuck at a percentage for hours. Pressing enter resumes the process.

**How to Reproduce the problem**
It doesn't happen every time, but most often during analyzing videos. They are typically large files (e.g. 540x720 for 200,000-400,000 frames).

**Comment by MMathisLab** [comment]
is it possible your computer is hibernating? We have never seen this behavior.

**Comment by rglanz** [comment]
Sorry for the late reply. No, the computer is not hibernating, it can occur while I'm actively working on the computer.

**Comment by MMathisLab** [comment]
Hmm... strange. Maybe take a video of this for us when it happens?

**Comment by rglanz** [comment]
I'm not sure a video would be very helpful, it essentially just displays the last line ran (e.g. iteration: xxxxxx loss: xxxx lr: xxxx) and the gpu is no longer being utilized (nvidia-smi shows gpu use at 0-1%).  When I hit enter in the conda prompt window, the gpu kicks back on and the function resumes.

It's happened throughout me having a couple different nvidia drivers/CUDA versions installed, so I'm going check into whether it's a hardware issue and I can update when I find out.

**Comment by MMathisLab** [comment]
okay thanks, we haven't seen this and no other reports of this, so perhaps a hardware issue... is your card overheating by chance?!

**Comment by rglanz** [comment]
That's my current thought. I am not very easy on my gpu....

**Comment by ConradAlting** [comment]
Dear Mr. and Ms. Mathis,

I‘m encountering the same problem as rglanz.
I managed to train a network for 600.000 iterations without any problems (that was  about 4 weeks ago). Whenever I start a new training now (no hardware changes, no changes on the software side of things, I.e. Anaconda, dlc-windowsGPU environment packages etc.), DLC stops using my GPU (GTX 1080 Amp! Extreme), the CPU usage for python.exe goes up from about 13% to 20% (i7 3930k liquid cooled, not overclocked).
I‘m out of ideas so far. I did a clean install of Windows 10, installed latest version of CUDA toolkit, tf 1.13.1 (per dlc-windowsGPU file provided on your Website, latest GeForce driver but to no avail.
As I’m away from my working horse til Sunday I’m not able to give more details on the specific versions used, so I‘ll come back to this post on Sunday to give you more informations.
I‘ll also attach a video with DLC running in a jupyter notebook and the taskmamager as well as nvidia-smi open so you might get a better idea of what’s going on.

Kind regards
Conrad

**Comment by j-diefenbach** [comment]
EDIT: I looked into it and noticed it also happened when I selected the terminal a certain way, so it appears it is due to text selection pausing code execution. Most likely I'm accidentally selecting the text when rearranging my desktop and don't notice until later. So I think the problem is solved and it may have been the case above, as hitting enter or even right arrow (as i just discovered) continues execution because the selected text gets unselected.

I believe I am having the same issue, and I wasn't able to see another issue that matches my problem like this one so I'm commenting here. Deeplabcut (seemingly) randomly pauses during anything from training to analysing videos. I don't recall whether this happens at the same frames (which could be an issue with the videos) since it happens sparsely (less than 10 times when running long videos). I am sure it is not hibernating as it will occur while using the computer still. I have seen it pause once using the graph and it simply goes straight down to 0% GPU usage (although I don't have a screenshot).

So far hitting enter solves the problem, and I can see the GPU Cuda graph on task manager go back up to 45-50% usage when I do so. 
Terminal and task manager during freeze:
<img width="1664" alt="bug1" src="https://github.com/user-attachments/assets/067f6e92-069a-47b2-ac7e-7ba94397be74" />
Immediately after hitting enter in console:
<img width="1664" alt="bug2" src="https://github.com/user-attachments/assets/50118819-903f-4a4d-8570-73e4ab3129ab" />


This is happening on DLC 3.0.0rc3, windows 10, I recently reinstalled pytorch and torchvision using `conda install pytorch torchvision pytorch-cuda=12.1 -c pytorch -c nvidia`
output from `nvcc -V`:
```
Cuda compilation tools, release 12.6, V12.6.20
Build cuda_12.6.r12.6/...
```

**Comment by MMathisLab** [comment]
Is your gpu over heating? There is nothing in the software that I think to explain any pauses, it sounds like a hardware issue or a corrupted image in training?

**Comment by j-diefenbach** [comment]
> EDIT: I looked into it and noticed it also happened when I selected the terminal a certain way, so it appears it is due to text selection pausing code execution. 

Not overheating or hardware / corruption issues, just due to the terminal. https://www.reddit.com/r/sysadmin/comments/118l8s0/am_i_crazy_or_does_terminal_progress_sometimes/
